### ISSUE TITLE
Could MoltenVK be automatically installed by Cargo?

### ISSUE THREAD

**Issue by mcclure** [vulkano-rs/vulkano#927]
Hello, I am new to both Rust and Vulkan. I tried out the Triangle example and it worked on Windows. Then I tried it on Mac and it failed because MoltenVK was not installed. Well of course not, I didn't install it.

But: Cargo was able to install all the *other* libraries it needed. I understand that there are some C based projects already in Cargo, for example libgit2 https://crates.io/crates/libgit2-sys/0.4.2 Would it be possible to make a Cargo package for MoltenVK, so that the requirement to externally install MoltenVK to use Vulkano could be waived, and we could get the versioning advantages of cargo for MoltenVK?

(I see there are a couple PRs right now related to this week's open source release of MoltenVK, but they seem to just be about making Vulkano work with an externally installed MoltenVK.)

**Comment by mitchmindtree** [comment]
@freesig and I are also very interested in this! I think @freesig is going to have a go at automating  this process in a standalone crate first, then we'll probably do a PR here and get some feedback.

One thing that I'm guessing will be necessary will be for this to be optional. @rukai perhaps having an opt-in `moltenvk_installer` feature for this would be enough?

@freesig maybe before getting started it would be worth listing all the individual steps here that the `moltenvk_installer` crate would perform at build time so we can get some feedback?

**Comment by rukai** [comment]
Yeah a feature sounds good, in the future we might also allow for gfx-rs as a backend via a similar feature.

**Comment by freesig** [comment]
Oh just saw this. I already built it, it's pretty much just a curl with some bash commands then you set the environment variables.
I'm having some issues with curl though. I'll post it here once it's working properly for some feedback

**Comment by freesig** [comment]
Just thought I'd post this here just incase anyone would like to use it. I will try and make a PR that integrates it into the build.rs aswell
https://crates.io/crates/moltenvk_deps
